### PR TITLE
get account id from afp api to use it in costLink for cloudhealth

### PR DESCRIPTION
**Pull request by mlefringhausen**
https://..../afp-api/latest/account?withid 
returns 
'"testaccount": {"id": "123456789012", "roles": ["testrole"]}'

### CODE REVIEW

File: js/service/userMessageService.js [hunk 1]
```diff
@@ -37,19 +37,34 @@ angular.module('afpClientApp').factory('userMessageService', [
...
-          'AccountName': accountName,
-          'Roles': []
-        };
+      angular.forEach(response, function (details, accountName) {
+        var account = "";
+        if ("id" in details) {
+          account = {
+            'AccountName': accountName,
+            'Roles': [],
+            'AccountID': details['id']
```
Comment (SebastianStiehl) [inline js/service/userMessageService.js]: Object key access is better written with the dot notation. `details.id`

File: js/service/userMessageService.js [hunk 2]
```diff
@@ -37,19 +37,34 @@ angular.module('afpClientApp').factory('userMessageService', [
...
+          };
 
-        angular.forEach(roles, function (role) {
-          account.Roles.push({
-            'Name': role,
-            'URLSuffix': accountName + '/' + role
+          angular.forEach(details['roles'], function (role) {
```
Comment (SebastianStiehl) [inline js/service/userMessageService.js]: Same here. Object key access is better written with the dot notation. `details.roles`

File: js/service/userMessageService.js [hunk 3]
```diff
@@ -37,19 +37,34 @@ angular.module('afpClientApp').factory('userMessageService', [
     function createAccountArray(response) {
       var result = [];
 
-      angular.forEach(response, function (roles, accountName) {
-        var account = {
-          'AccountName': accountName,
-          'Roles': []
-        };
+      angular.forEach(response, function (details, accountName) {
+        var account = "";
```
Comment (SebastianStiehl) [inline js/service/userMessageService.js]: The account setup can be done in the beginning and filled in later.
```
var account = {
  'AccountName': accountName,
  'Roles': []
};

...

account.AccountID = details.id;
```

**Review by SebastianStiehl — CHANGES_REQUESTED** [review]

**Comment by mlefringhausen** [comment]
I did the requested changes and will create a new pull request

**Pull request closed without merging**